src/components/MyBooks.js

```diff
@@ -1,10 +1,113 @@
-import React from 'react'
+import {Auth, API, graphqlOperation } from 'aws-amplify'
+import React,{useEffect, useState} from 'react'
+import { toast } from 'react-toastify';
+
+import { deleteBook } from '../graphql/mutations'
+import { listBooks } from '../graphql/queries'
+import { useAuth } from './contexts/AuthContext';
+import MyBooksItem from './MyBooksItem'
+
 
 const MyBooks = () => {
-    return (
-        <div>   
-            This is my books
+    const [books, setBooks] = useState([])
```

Comment:
One good pattern here would be to use useReducer() instead of useState(). That way you can decouple the logic from this component. Have them separate out in the reducer files. Then you can dispatch actions such as add, update, remove books.